packages/opentelemetry-sdk-trace-web/src/utils.ts

```diff
@@ -125,6 +129,139 @@ export function addSpanNetworkEvents(
   }
 }
 
+function _getBodyNonDestructively(body: ReadableStream) {
+  // can't read a ReadableStream without destroying it
+  // on most platforms, we CAN tee the body stream, which lets us split it,
+  // but that still locks the original stream, so we end up needing to return one of the forks.
+  //
+  // some (older) platforms don't expose the tee method and in that scenario, we're out of luck;
+  //   there's no way to read the stream without consuming it.
+  if (!body.tee) {
+    return {
+      body,
+      length: Promise.resolve(null),
+    };
+  }
+
+  const [bodyToReturn, bodyToConsume] = body.tee();
```

Comment:
So, somewhat embarrassingly I also started working on adding support for this (completely forgetting to re-check whether anyone else had started this since the last time I looked into it), but rather than make a competing PR maybe I can just share some thoughts.

Based on what I learned in my approach, you can avoid the overhead of `ReadableStream.tee()` by instead creating a [`TransformStream`](https://developer.mozilla.org/en-US/docs/Web/API/TransformStream) which counts size chunk by chunk, and using `ReadableStream.pipeThrough()` to send data through that stream.

For example:
```typescript
// ... in patched fetch/Request when there is a ReadableStream body
const transform = new ByteCountingTransformStream(); // counts chunk length using its `transform` method
init.body = readable.pipeThrough(transform);
// ... exposing a function somewhere to read the size from the transformer
init.body.byteLength = async () => { return await transform.byteLength() };
// ... reading it later when applicable
const bytes = await init.body.byteLength();
```

Reply:
This is a good suggestion, thanks! I implemented it here https://github.com/open-telemetry/opentelemetry-js/pull/4706/commits/102d1286c6187eb1b24735f85e3159c7aae6d5f2

Reply:
@MustafaHaddara thanks for taking the suggestion and implementing it! I'm glad that it was useful/helpful.